src/main/java/spoon/reflect/declaration/CtElement.java

```diff
@@ -249,11 +249,15 @@
 
 	/**
 	 * Gets the first parent that matches the given type.
+	 *
+	 * @return null when a matching parent was not found or this element doesn't have a parent
```

Comment:
Sorry, I wasn't clear at all here. `@return` should describe the return value as a whole, so it should say something like so:

```suggestion
	 * @return The closest matching parent, or null if no match is found
```

To me, "doesn't have a parent" is included in "no match is found", but if you think that's not clear feel free to explicitly state that case as well.

Reply:
```suggestion
	 * @return the nearest matching parent; null if no match is found or this element has no parent
```

`getRoleInParent` javadoc has a "no parent" situation description, so for the consistency sake I'm including it here. Any further edits?

Reply:
> getRoleInParent javadoc has a "no parent" situation description, so for the consistency sake I'm including it here

Good observation, I agree.

> Any further edits?

Same edit for the other `@return`, but nothing other than that. When those are done, we call upon an integrator to make the final check.